client/jetpack-onboarding/steps/stats.jsx

```diff
@@ -0,0 +1,132 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Fragment } from 'react';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import DocumentHead from 'components/data/document-head';
+import FormattedHeader from 'components/formatted-header';
+import PageViewTracker from 'lib/analytics/page-view-tracker';
+import Tile from 'components/tile-grid/tile';
+import TileGrid from 'components/tile-grid';
+import { addQueryArgs } from 'lib/route';
+import { getUnconnectedSiteUrl } from 'state/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { JETPACK_ONBOARDING_STEPS as STEPS } from '../constants';
+import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions';
+
+class JetpackOnboardingStatsStep extends React.Component {
+	componentDidUpdate() {
+		const { action, activatedStats, isConnected } = this.props;
+
+		if ( isConnected && activatedStats === false && action === 'activate_stats' ) {
+			this.activateStats();
+		}
+	}
+
+	handleActivateStats = () => {
+		this.props.recordJpoEvent( 'calypso_jpo_stats_clicked' );
+
+		if ( ! this.props.isConnected ) {
+			return;
+		}
+
+		this.activateStats();
+	};
+
+	handleStatsNextButton = () => {
+		this.props.recordJpoEvent( 'calypso_jpo_stats_next_clicked' );
+	};
+
+	activateStats() {
+		this.props.saveJetpackOnboardingSettings( this.props.siteId, {
+			stats: true,
+		} );
+	}
+
+	renderSuccess() {
+		const { getForwardUrl, translate } = this.props;
+		const headerText = translate( 'Success! Jetpack is now collecting valuable stats.' );
+
+		return (
+			<Fragment>
+				<FormattedHeader headerText={ headerText } />
```

Comment:
Nit, I'd inline `headerText`. Mostly since it's only used here, and it's (if slightly) easier to visualize the component.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/22621/commits/7da139cd020ad29e14870d9de9992d07ff98f1e7